clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -299,17 +299,28 @@ public void ensurePartitionAssignment() {
             ensureActiveGroup();
     }
 
-    @Override
-    public void close() {
-        // commit offsets prior to closing if auto-commit enabled
-        while (true) {
-            try {
-                maybeAutoCommitOffsetsSync();
-                return;
-            } catch (WakeupException e) {
-                // ignore wakeups while closing to ensure we have a chance to commit
-                continue;
+    public void close(long timeoutMs) {
+        if (timeoutMs < 0)
+            throw new IllegalArgumentException("Timeout must be non-negative");
+
+        long now = time.milliseconds();
+        long deadline = now + timeoutMs;
+        if (deadline < 0) deadline = Long.MAX_VALUE;
```

Comment:
I was originally thinking if ```close()``` is called we will just send a leave group request and wait on the response forever, and relying on the request timeout to return if the coordinator is ever not responding, i.e. calling ```client.poll(request)``` instead of ```client.poll(request, timeout = Long.MAX)```.